clients/drcachesim/tracer/output.cpp

```diff
@@ -122,25 +122,33 @@ reached_traced_instrs_threshold(void *drcontext)
         dr_mutex_unlock(mutex);
         return;
     }
-    // We've reached the end of our window.
-    // We do not attempt a proactive synchronous flush of other threads'
-    // buffers, relying on our end-of-block check for a mode change.
-    // (If -retrace_every_instrs is not set and we're not going to trace
-    // again, we still use a counting mode for simplicity of not adding
-    // yet another mode.)
-    NOTIFY(0, "Hit tracing window #%zd limit: disabling tracing.\n",
-           tracing_window.load(std::memory_order_acquire));
-    // No need to append TRACE_MARKER_TYPE_WINDOW_ID: the next buffer will have
-    // one in its header.
-    // If we're counting at exit time, this increment means that the thread
-    // exit entries will be the only ones in this new window: but that seems
-    // reasonable.
-    tracing_window.fetch_add(1, std::memory_order_release);
-    // We delay creating a new ouput dir until tracing is enabled again, to avoid
-    // an empty final dir.
-    DR_ASSERT(tracing_mode.load(std::memory_order_acquire) == BBDUP_MODE_TRACE);
-    tracing_mode.store(BBDUP_MODE_COUNT, std::memory_order_release);
-    cur_window_instr_count.store(0, std::memory_order_release);
+    ptr_int_t mode = tracing_mode.load(std::memory_order_acquire);
+    if (mode == BBDUP_MODE_L0_FILTER) {
+        NOTIFY(0, "Hit tracing window #%zd filter limit: switching to full trace.\n",
```

Comment:
This might be mistaken to mean that each tracing window has its own filter limit. Could replace it with: "Hit filter limit while at window". Same below.

Reply:
I kept it consistent with the message that is shown when full trace is done. I will update both.

```
Hit tracing window #0 filter limit: switching to full trace.
Hit tracing window #0 limit: disabling tracing.
```

Reply:
> This might be mistaken to mean that each tracing window has its own filter limit

Looks like I was wrong and each tracing window indeed has its own filter limit. In that case it makes sense to add the window number to this log.

Reply:
Yes each window has a filter phase. I thought you meant that it sounded like each window has a different limit.